ibis/tests/sql/test_compiler.py

```diff
@@ -1211,44 +1211,33 @@ def test_fuse_projections(self):
         table2_filtered = table2[pred]
 
         f2 = (table2['foo'] * 2).name('qux')
-        f3 = (table['foo'] * 2).name('qux')
 
         table3 = table2.projection([table2, f2])
 
         # fusion works even if there's a filter
         table3_filtered = table2_filtered.projection([table2, f2])
 
-        expected = table[table, f1, f3]
-        expected2 = table[pred][table, f1, f3]
-
-        assert table3.equals(expected)
-        assert table3_filtered.equals(expected2)
-
-        ex_sql = """SELECT *, `foo` + `bar` AS `baz`, `foo` * 2 AS `qux`
-FROM tbl"""
+        ex_sql = """\
+SELECT *, `foo` * 2 AS `qux`
+FROM (
+  SELECT *, `foo` + `bar` AS `baz`
+  FROM tbl
+) t0"""
```

Comment:
It's sort of a bummer to miss the projection fusion opportunity here, but this SQL more clearly reflects the intent of the user code in this case

Reply:
I have plans to bring this kind of stuff back into ibis, but we have to get to place where we generate the correct code before bringing it back :)